hat/assets/js/apps/Iaso/domains/orgUnits/components/TreeView/requests.ts

```diff
@@ -0,0 +1,199 @@
+import { UseQueryResult } from 'react-query';
+import { errorSnackBar } from '../../../../constants/snackBars';
+import { getRequest } from '../../../../libs/Api';
+import { useSnackQuery } from '../../../../libs/apiHooks';
+import { enqueueSnackbar } from '../../../../redux/snackBarsReducer';
+import { dispatch } from '../../../../redux/store';
+import { OrgUnit, OrgUnitStatus } from '../../types/orgUnit';
+
+const getValidationStatus = (statusSettings: OrgUnitStatus[]): string => {
+    return statusSettings
+        .map(status => `&validation_status=${status}`)
+        .join('');
+};
+
+const baseApiUrl = '/api/orgunits/tree/';
+
+export const getChildrenData = async (
+    id: string,
+    statusSettings: OrgUnitStatus[],
+): Promise<OrgUnit[]> => {
+    try {
+        const response = await getRequest(
+            `${baseApiUrl}?parent_id=${id}&ignoreEmptyNames=true${getValidationStatus(
+                statusSettings,
+            )}`,
+        );
+        return response.map((orgUnit: any) => ({
+            ...orgUnit,
+            id: orgUnit.id.toString(),
+        }));
+    } catch (error: unknown) {
+        dispatch(
+            enqueueSnackbar(errorSnackBar('getChildrenDataError', null, error)),
+        );
+        console.error('Error while fetching Treeview item children:', error);
+        throw error;
+    }
+};
+
+const makeUrl = (
+    id?: string | number,
+    type?: string,
+    statusSettings: OrgUnitStatus[] = ['VALID'],
+): string => {
+    const validationStatusString = getValidationStatus(statusSettings);
+    let typePrefix;
+
+    switch (type) {
+        case 'version':
+            typePrefix = 'version';
+            break;
+        case 'source':
+            typePrefix = 'data_source_id';
+            break;
+        default:
+            break;
+    }
+
+    const idParam = id && typePrefix ? `&${typePrefix}=${id}` : '';
+    return `${baseApiUrl}?ignoreEmptyNames=true${idParam}${validationStatusString}`;
```

Comment:
Unless the order of the params is important, I think we could use early returns in the switch statement. 
I prefer when we can avoid re-assigning value, but I can also live with it

```suggestion
    const defaultUrl = `${baseApiUrl}?ignoreEmptyNames=true${validationStatusString}`;
    if(!id) return defaultUrl;
    switch (type) {
        case 'version':
            return `${defaultUrl}&version=${id};
        case 'source':
            return `${defaultUrl}&data_source_id=${id};
        default:
            return  defaultUrl;
    }
```